packages/widgets/src/tabbar.ts

```diff
@@ -1731,11 +1807,14 @@ export namespace TabBar {
      * @returns The ARIA attributes for the tab.
      */
     createTabARIA(data: IRenderData<any>): ElementARIAAttrs | ElementBaseAttrs {
-      return {
+      const ariaAttributes: { [k: string]: string } = {
         role: 'tab',
-        'aria-selected': data.current.toString(),
-        tabindex: '0'
+        'aria-selected': data.current.toString()
       };
+      if (data.tabIndex !== undefined) {
+        ariaAttributes.tabindex = `${data.tabIndex}`;
```

Comment:
Having `tabindex="-1"` as default would probably make sense for a tab.

I'm not quite sure about the difference between `tabindex="-1"` and tabindex not set.
It seems that the behavior is not the same for input elements and other elements.
In the case of the tab, not setting the `tabindex` would make it not focusable with tabulation, nor 'programatically'. I'm not sure how useful it can be...

Reply:
> I'm not quite sure about the difference between `tabindex="-1"` and tabindex not set.
> It seems that the behavior is not the same for input elements and other elements.

when `tabindex` is not defined, interactive elements automatically receive tab focus, while non-interactive elements don't. so if you wan't to avoid a tab on an interactive widget you'd use `tabindex=-1`. if you want to add a non-interactive dom element to the tab order then you'd use `tabindex=0`. 

the [ARIA authoring guide example for this pattern](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/) uses `button`s for the tab element which means they need control what doesn't get tabbed to; the implementation needs to choose one tab element to have `tabindex=0`, or undefined, and the rest are `tabindex=-1`. if the tab elements are made of non-interactive elements then the task is to make elements appear in the dom order by adding `tabindex=0` to one element.

> In the case of the tab, not setting the `tabindex` would make it not focusable with tabulation, nor 'programatically'. I'm not sure how useful it can be...

hopefully folks don't mess with tab order. that should be private. most recommendations say to avoid `tabindex>0`.

Reply:
fixed in https://github.com/jupyterlab/lumino/pull/612/commits/9d395d8c8f5dbe2194644636c8e251a1d6285aad